tests/linear_model/test_glm.py

```diff
@@ -203,3 +219,121 @@ def test_logistic_predict_proba_shape():
     lr.fit(X, y)
     prob = lr.predict_proba(X)
     assert prob.shape == (100, 2)
+
+
+@pytest.mark.parametrize(
+    "reg_type,data_generator",
+    [
+        (LinearRegression, "single_chunk_regression"),
+        (LogisticRegression, "single_chunk_classification"),
+        (PoissonRegression, "single_chunk_count_classification"),
+    ],
+)
+def test_model_coef_dask_numpy(reg_type, data_generator, request):
+    """Tests that models return same coefficients and intercepts with array types"""
+    X, y = request.getfixturevalue(data_generator)
+    np_mod, da_mod = reg_type(fit_intercept=True), reg_type(fit_intercept=True)
+    da_mod.fit(X, y)
+    np_mod.fit(X.compute(), y.compute())
+    da_coef = np.hstack((da_mod.coef_, da_mod.intercept_))
+    np_coef = np.hstack((np_mod.coef_, np_mod.intercept_))
+
+    rel_error = LA.norm(da_coef - np_coef) / LA.norm(np_coef)
+    assert rel_error < 1e-8
+
+
+@pytest.mark.parametrize("solver", ["newton", "gradient_descent", "lbfgs"])
+@pytest.mark.parametrize("fit_intercept", [True, False])
+@pytest.mark.parametrize(
+    "reg_type, skl_reg_type, skl_params, data_generator",
+    [
+        (LinearRegression, slm.LinearRegression, {}, "medium_size_regression"),
```

Comment:
Style nit: maybe this would be a bit simpler and a bit shorter:

``` python
# fmt: off
@pytest.mark.parametrize("est, skl_params, data", [
    ("LinearRegression", {}, medium_size_regression()),
    ("LogisticRegression", {"penalty": None}, single_chunk_classification()),
])
# fmt: on
def test_model_against_sklearn(est, skl_params, data):
    X, y = data
    EstDask = getattr(dask_ml.linear_regression, est)
    EstSklearn = getattr(scikit_learn.linear_model, est)
    dask_est = EstDask(...)
    skl_est = EstSklearn(...)
    ...
```

Reply:
I adopted this set of tests with one minor difference -- when combining the tests for coefficient as well as `PoissonRegression`, there is actually a naming difference between `dask-ml` and `scikit-learn`. I have added a second parameter and a `try-except` block to handle that difference.